packages/sanity/src/core/store/_legacy/document/document-pair/serverOperations/patch.ts

```diff
@@ -29,25 +29,14 @@ export const patch: OperationImpl<[patches: any[], initialDocument?: Record<stri
 
     const patchMutation = draft.patch(patches)
 
-    if (snapshots.published) {
-      draft.mutate([
-        // If there's no draft, the user's edits will be based on the published document in the form in front of them
-        // so before patching it we need to make sure it's created based on the current published version first.
-        draft.createIfNotExists({
-          ...initialDocument,
-          ...snapshots.published,
-          _id: idPair.draftId,
-          _type: typeName,
-        }),
-        ...patchMutation,
-      ])
-      return
-    }
+    // If there's no draft, the user's edits will be based on the published document in the form in front of them
+    // so before patching it we need to make sure it's created based on the current published version first.
     const ensureDraft = snapshots.draft
       ? []
       : [
           draft.create({
```

Comment:
There is a problem here. We cannot create a draft for an existing published document; we would instead need to edit the published document.

Because we removed the handling for when `snapshot.published` exists, we now often attempt to create a draft for an existing published document, causing Studio to crash with the following error:

```
Error: there is already a published document with that ID
```

I'm currently trying to reason about how we should proceed.